cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/detector/AnomalyDetector.java

```diff
@@ -441,7 +457,7 @@ private void fixAnomalyInProgress(AnomalyType anomalyType) throws Exception {
             if (isReadyToFix) {
               LOG.info("Fixing anomaly {}", _anomalyInProgress);
               fixStarted = _anomalyInProgress.fix();
-              String optimizationResult = _anomalyInProgress.optimizationResult(false);
+              String optimizationResult = ((KafkaAnomaly)_anomalyInProgress).optimizationResult(false);
```

Comment:
This casting seems a little flaky -- can we avoid this by making the anomaly interface (i.e. relevant return types, supported functions) even more generic?

Reply:
moved `String optimizationResult(boolean isJson)` method to `Anomaly` interface.